pallets/msa/src/lib.rs

```diff
@@ -610,12 +610,18 @@ pub mod pallet {
 			Ok(())
 		}
 
-		/// Adds a given `new_key` to `msa_id` of the account signing `msa_owner_proof`, which must match the MSA in `add_key_payload`.
-		/// The `new_key` must sign the `add_key_payload` to authorize the addition.
+		/// Adds a new public key to an existing Message Source Account (MSA). This functionality enables the MSA owner to manage multiple keys
+		/// for their account or rotate keys for enhanced security.
 		///
-		/// # Remarks
-		/// * Origin can be same as msa owner.
-		/// * Signatures should be over the [`AddKeyData`] struct
+		/// The `origin` parameter represents the account from which the function is called and can be either the MSA owner's account or a delegated provider's account,
```

Comment:
> The `origin` parameter represents the account from which the function is called

While that's technically true, the Origin is the same as any other extrinsic Origin, the submitter of the extrinsic call.

The extrinsic is designed to allow either the MSA Owner (if they pay token for it) or the MSA's Provider to be able to submit the extrinsic, so that new keys can be added via a Provider's dApp (so they _don't_ pay token for it)